### DISCUSSION TITLE
Renovate support

### DISCUSSION THREAD

**Discussion by patrickleet** [crossplane/crossplane#4319] [Q&A] [answered]
### What problem are you facing?

I see this repo is using renovate! I'd love for renovate to support crossplane to help keep up to date:

1. My providers
2. Configs from the marketplace
3. Helm releases used in my Composites

### How could Crossplane help solve your problem?

This isn't a crossplane, issue, per se, but I'm sure some collaboration between crossplane and renovate would go along way.

I brought this up in a discussion in renovate's repo as well. They provided a form to fill out to add a new package manager request:
https://github.com/renovatebot/renovate/discussions/22363 

I can fill out what I know but maybe a maintainer could provide better info than me.

Just wanted to see if I could get some traction started on this on both sides!

Also, if anyone has solutions, or partial solutions to this already, using regexps or something, please share them here!

**Comment by phisco** [comment]
This could make even more sense for ourselves, given that with the new E2Es we are going to have many more examples and it would be nice to have them updated without the need to annotate each manifest. I'll try to look into this, thanks @patrickleet 🙏

**Comment by phisco** [comment]
It looks like it's mostly going to require us to write an image extractor (see [this](https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/docker-compose/extract.ts#L63) and other examples for already implemented managers) for the resources we define that have references to OCI images, so Provider and Configuration, we could think of also supporting composition functions in Compositions. I would say introspecting the bases in the resources will be out of scope and should be left to the users to use regexes or annotations to have that.

**Comment by turkenh** [comment]
It looks like this is not something to implement in Crossplane repo, rather looking for a guideline or some examples on how to configure renovate for provider and configuration images. So, I believe this issue better fits a discussion.

**Comment by danports** [comment]
I have a fairly basic Crossplane setup so here's what I use to renovate my provider versions (from my Flux repo's `renovate.json`) - I guess this could work for configuration versions too, given a suitable `fileMatch`:
```json
"regexManagers": [
    {
        "fileMatch": [
            "crossplane/provider-\\w+\\.yaml$"
        ],
        "matchStrings": [
            "package:\\s*\"(?<depName>.*?):(?<currentValue>.*?)\""
        ],
        "datasourceTemplate": "docker"
    }
]
```

(The only downside with this is that you can't pull release notes from Docker - you could try to use the `github-releases` datasource, but if the GitHub project path doesn't exactly match the Docker repository path, that won't work - I'm looking at you, `crossplanecontrib` vs. `crossplane-contrib`... 😒 )

I contributed the initial version of the Flux manager to Renovate and thought about writing a Crossplane manager too, but decided not to because I could get away with just a simple regex manager configuration. If someone were to contribute a Crossplane manager to Flux, I think the easiest approach would be to clone the Flux manager (since both managers require reading YAML manifests) and then tailor it to Crossplane's resources.

**Comment by phisco** [comment]
FYI: I opened a PR to add support for crossplane packages: https://github.com/renovatebot/renovate/pull/25896

**Reply by phisco** [reply, marked as answer]
FYI: [support](https://github.com/renovatebot/renovate/pull/25896) was merged 🎉
